frontend/src/libs/packages/store/store.package.ts

```diff
@@ -38,13 +39,14 @@ class Store {
         auth: authReducer,
         users: usersReducer,
       },
-      middleware: (getDefaultMiddleware) => {
-        return getDefaultMiddleware({
+      middleware: (getDefaultMiddleware) => [
+        ...getDefaultMiddleware({
```

Comment:
as far as I know, using the spread operator to add middleware can cause problems with dispatch function being typed correctly. Maybe it is better to use concat instead of spread operator? Therefore, there will be no problem with inferring the dispatch function in the future.

Moreover, it is a suggested way to define middleware from redux toolkit docs: https://redux-toolkit.js.org/usage/usage-with-typescript/#correct-typings-for-the-dispatch-type

Reply:
please use concat here as @HIMDanny said

Reply:
In pr #62 on mobile after adding concat the linter error appears, maybe the same will happen here

Reply:
I have tried but I'm not able to the handleError middleware within the concat method in your Redux store setup presents an error message that indicates a type mismatch in the middleware array. The problem seems to arise from the incompatibility of types. since the type includes more middleware than just ThunkMiddleware, which is what is expected, I have tried implementing custom types and searched for solutions but nothing has solved it yet

Reply:
I have tried but I'm not able to, the handleError middleware within the concat method in the Redux store setup presents an error message that indicates a type mismatch in the middleware array. The problem seems to arise from the incompatibility of types. since the type includes more middleware than just ThunkMiddleware, which is what is expected, I have tried implementing custom types and searched for solutions but nothing has solved it yet

Reply:
ok, then let's leave it for now